### PR TITLE
#12 two types of data generator with three arg parsers

### LINKED ISSUE #12: data_builded should not be called before inference
It is quite disadvantageous that the data_builded script has to be called before inference. Then it will work only with the data from the ASIP (note P instead of D) dataset. However we will use it also for other data from Sentinel-1 and AMSR-2. Therefore the model should be applicable to any input with such data. 

One way to apply a model without a builder script is to make a new generator which accepts an in-memory object as input, instead of a list of NPZ files. The Archive class already has the functionality to read everything into memory (and also to write NPZ files, which is relevant only for training dataset). Now a new DataGenerator should be developed to take Archive object as input.
Then Archive becomes not a proper class as it mixes operations on archive (multiple files) and as single dataset. So it should be split into two (e.g. Archive and Dataset) and then the new DataGenerator should take only a Dataset object.

Later (another issue), in order to adapt the generator to other input data, we will develop a class that can read Sentinel-1 and AMSR2 from two different files, collocate them on the same grid, create an object with the same interface as the Dataset above and use it either for building another training dataset or for inference.

### PR DESCRIPTION
**Pull request by azamifarda**
close #12 
@akorosov 
I have deloped a code and the refactored both the new code and the old ones a lot. SO what you see in this PR is the result of lots of refactoring.

### CODE REVIEW

**Review by akorosov — CHANGES_REQUESTED** [review]
It became too convoluted and too difficult to change (and also understand). E.g. if another model architecture should be tested the Configure class has to be changed. But the base classes should not be changed by users. Besides train_model was just an example script, which should be easy to change (or even convert to a notebook)

* Split apply_or_train_mode.py into two scripts: train_model.py, apply_model.py
* Move definition of FileBasedConfigure to train_model.py 
* Move create_model from Configure to root of utility (it is just an example)
* Move train_model and other methods related to training only out of  Configure to train_model.py (maybe to FileBasedConfigure)
* Move MemoryBasedConfigure to apply_model.py
* Move methods related only application of a model out of base configure to MemoryBasedConfigure
* Exclude parameters from utility.read_input_params that are related to apply/train and file/memory selection. Also remove unnecessary parameters from Archive (e.g.  memory_mode)

File: asip_v2/build_dataset.py [hunk 1]
```diff
@@ -17,19 +17,10 @@ def main():
...
-            for cls_ in [SarBatches, OutputBatches, Amsr2Batches]:
-                obj = cls_(archive_)
-                obj.pad_and_batch(fil)
-                archive_.PROP.update(obj.calculate_variable_ML())
-                del obj
-            del archive_.mask_batches_amsr2, archive_.mask_batches
-            # saving section
-            archive_.write_scene_files_and_reset_archive_PROP()
-            archive_.update_processed_files(i)
+        archive_.calculate_PROP_of_archive(fil, filename)
```
[the remark this answers is not in the document]
Comment (azamifarda) [reply asip_v2/build_dataset.py]: Done

File: asip_v2/build_dataset.py [hunk 2]
```diff
@@ -17,19 +17,10 @@ def main():
...
+        # saving section
+        archive_.write_scene_files_and_reset_archive_PROP()
```
[the remark this answers is not in the document]
Comment (azamifarda) [reply asip_v2/build_dataset.py]: Done

File: asip_v2/utility.py [hunk 3]
```diff
@@ -1,76 +1,82 @@
...
 from tensorflow.keras import layers
 from tensorflow.keras.models import Model
 
 from archive import Archive
-from data_generator import DataGenerator
 
+def parse_common_args():
+    """Instantiates and creates common arguments of parser which works with 'argparse' of python."""
+    parser = argparse.ArgumentParser(description='Process the arguments of script')
+    parser.add_argument()
```
Comment (akorosov) [inline asip_v2/utility.py:16]: Is that used? I couldnt find...

File: asip_v2/train_model.py [hunk 4]
```diff
@@ -1,48 +1,182 @@
...
+    #    #x = layers.Conv2DTranspose(filters=16, kernel_size=3, strides=2, padding='same')(x)
+    #    #x = layers.Conv2DTranspose(filters=8, kernel_size=3, strides=2, padding='same')(x)
+    #    x = layers.Conv2D(filters=1, kernel_size=1, strides=1, padding='same')(x)
+    #    self.model = Model(
+    #        inputs=[input_, input_2], outputs=x)
+    #    opt = tf.keras.optimizers.Adam(
+    #        learning_rate=0.001, beta_1=0.9, beta_2=0.999, epsilon=1e-07, amsgrad=False
+    #        )
+    #    self.model.compile(optimizer=opt, loss='mse', metrics=['accuracy'])
+
```
Comment (akorosov) [inline asip_v2/train_model.py:86]: Please remove commented lines

File: README.md [hunk 5]
```diff
@@ -4,79 +4,108 @@ This repository provides the facilities for ASIP v2 data [(webpage)](https://dat
...
-```python
-python build_dataset.py /absolute/path/to/the/folder/of/input_files
-```
-
-This command will create a folder named **output** in the folder that contains the `build_dataset.py` and write all the output files into it. This command is executable with two different modes, name Training and Inference which are determined by ABSENCE or PRESENCE of `-i` in the commandline.
-
->  HINT: The folder containing input files must have all of the **.nc** files without any subfoldering
-
-The output address as well as some other parameters for this process is configurable. They can be given to the command above as an argument of command. Table below shows the detail of them:
+# Usage
```
Comment (akorosov) [inline README.md:15]: Readme is outdated

**Review by akorosov — CHANGES_REQUESTED** [review]
Please update readme wrt CLI parameters + few minor things in the comments
It is not very nice to add testing inside this PR

File: asip_v2/utility.py (hunk 3 above)
Comment (azamifarda) [reply asip_v2/utility.py:16]: Sorry, I forgot to remove that during development.

Comment (azamifarda) [reply asip_v2/utility.py:16]: Done in the next commit

File: asip_v2/train_model.py (hunk 4 above)
Comment (azamifarda) [reply asip_v2/train_model.py:86]: Done in the next commit.

**Review by akorosov — APPROVED** [review]

**Pull request merged**